File: men_clothes-backend/shopapp/src/main/java/com/project/shopapp/models/Coupon.java
package com.project.shopapp.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "coupons")
@Data//toString
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Coupon extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(name = "discount", length = 50)
    private float discount;

    @Column(name = "content", length = 50)
    private String content;

    @Column(name = "codecp")
    private String codecp;

    @Column(name = "startday")
    private LocalDateTime startday;

    @Column(name = "endday")
    private LocalDateTime endday;
}

File: men_clothes-angular/src/app/models/coupon.ts
export class Coupon {
  category_id: number;
  discount: number;
  content: string;
  codeCp: string;
  startday: string;
  endday: string;

  constructor(
    category_id: number = 0,
    discount: number = 0,
    content: string = '',
    codeCp: string = '',
    startday: string = '',
    endday: string = ''
  ) {
    this.category_id = category_id;
    this.discount = discount;
    this.content = content;
    this.codeCp = codeCp;
    this.startday = startday;
    this.endday = endday;
  }
}